### ISSUE TITLE
admin check index failed due to tidb_mem_quota_query limit 

### ISSUE THREAD

**Issue by Lily2025** [pingcap/tidb#46089]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1、add index for table
2、admin check index

### 2. What did you expect to see? (Required)
admin check index success

### 3. What did you see instead (Required)
admin check index failed with below error

Error 1105 (HY000): Your query has been cancelled due to exceeding the allowed memory limit for a single SQL query. Please try narrowing your query scope or increase the tidb_mem_quota_query limit and try again.[conn=0]

### 4. What is your TiDB version? (Required)
./tidb-server -V
 Release Version: v7.4.0-alpha
Edition: Community
Git Commit Hash: 0f41cfe2a8e71447f8c381f1b3380b82437f0b34
Git Branch: heads/refs/tags/v7.4.0-alpha
UTC Build Time: 2023-08-10 13:54:56
GoVersion: go1.21.0
Race Enabled: false
Check Table Before Drop: false
Store: unistore

**Comment by ywqzzy** [comment]
Currently the check index query may consume many memory, we can set `tidb_mem_quota_query` to higher value to check if the query can run successfully.

**Comment by bb7133** [comment]
I don't think this is a bug, I changed the type of this issue to 'enhancement' since we hope the memory consumption of `ADMIN CHECK INDEX` syntax can be lower.

**Comment by tiancaiamao** [comment]
I remember @wjhuang2016 had some ideas on the enhancement of 'admin check table', not sure how is it going now.
